src/client/pages/Offer/Introduction/DetailsModal/Details/components/DetailInput.tsx

```diff
@@ -0,0 +1,372 @@
+  masks,
+  inputTypes,
+  InputGroup,
+  InputGroupDeleteButton,
+} from 'components/inputs'
+import { MarketLabel } from 'l10n/locales'
+import { useTextKeys } from 'utils/textKeys'
+import { QuoteInput } from '../../types'
+import { ContentColumnTitle, ContentColumnTitleButton } from './Details.styles'
+
+interface DetailInputProps {
+  field?: CoreInputFieldProps
+  formikProps: FormikProps<QuoteInput>
+  name: string
+}
+
+export const DetailInput: React.FC<DetailInputProps &
+  GenericFieldHTMLAttributes> = ({ field, formikProps, name: formikName }) => {
+  const textKeys = useTextKeys()
+
+  return field ? (
+    <InputField
+      label={textKeys[field.label]()}
+      placeholder={textKeys[field.placeholder ? field.placeholder : '']()}
+      name={formikName}
+      mask={field.mask}
+      type={field.type}
+      options={field.options?.map((o) => ({
+        label: textKeys[o.label](),
+        value: o.value,
+      }))}
+      showErrorMessage={false}
+      errors={getIn(formikProps.errors, formikName)}
+      touched={getIn(formikProps.touched, formikName)}
+      autoComplete="off"
+      onChange={(e: React.ChangeEvent<HTMLInputElement>) => {
+        const value = field.mask
+          ? field.mask.sanitize(e.target.value)
+          : e.target.value
+
+        if (field.type === 'number') {
+          formikProps.setFieldValue(formikName, value && parseInt(value, 10))
+        } else {
+          formikProps.setFieldValue(
+            formikName,
+            /^(true|false)/.test(value) ? JSON.parse(value) : value,
+          )
+        }
+      }}
+    />
+  ) : null
+}
+
+type TextInputProps = {
+  name: string
+  label: string
+  formikProps: FormikProps<QuoteInput>
+}
+
+export const TextInput: React.FC<TextInputProps> = ({
+  name,
+  label,
+  formikProps,
+}) => (
+  <DetailInput
+    name={name}
+    field={{
+      label: label,
+      placeholder: '',
+    }}
+    formikProps={formikProps}
+  />
+)
+
+type BooleanInputProps = {
+  name: string
+  label: string
+  formikProps: FormikProps<QuoteInput>
+}
+
+export const BooleanInput: React.FC<BooleanInputProps> = ({
+  name,
+  label,
+  formikProps,
+}) => (
+  <DetailInput
+    name={name}
+    field={{
+      label: label,
+      placeholder: '',
+      options: [
+        { label: 'YES', value: 'true' },
+        { label: 'NO', value: 'false' },
+      ],
+    }}
+    formikProps={formikProps}
+  />
+)
+
+type AreaInputProps = {
+  name: string
+  label: string
+  formikProps: FormikProps<QuoteInput>
+}
+
+export const AreaInput: React.FC<AreaInputProps> = ({
+  name,
+  label,
+  formikProps,
+}) => (
+  <DetailInput
+    name={name}
+    field={{
+      label: label,
+      placeholder: '',
+      mask: masks.area,
+      type: inputTypes.number,
+    }}
+    formikProps={formikProps}
+  />
+)
+
+type HomeOwnershipTypeInputProps = {
+  name: string
+  formikProps: FormikProps<QuoteInput>
+}
+
+enum HomeOwnershipType {
+  Rent = 'RENT',
+  Own = 'OWN',
+}
```

Comment:
Can't we get this from GraphQL client module?

Reply:
This is intentional. These fields including `HomeOwnershipTypeInput` component are meant to be simple, cross market form components. This specific component works for NO and DK. Unfortunately, the types in GraphQL client are market specific.